build-pipeline.yml

```diff
@@ -10,23 +10,42 @@
 
 parameters:
   # List of Python packages that need to be pip installed for testing.
-  extra_test_dependencies: []
+  - name: extra_test_dependencies
```

Comment:
Azure Pipelines now support fully type parameters for templates, so we'll document properly the supported parameters!